content/post/kubedb-v2024.11.18/index.md

```diff
@@ -0,0 +1,1269 @@
+      kind: Issuer # Issuer type (Issuer or ClusterIssuer)
+      apiGroup: "cert-manager.io"
+    certificates:
+      - alias: client
+        subject:
+          organizations:
+            - mssqlserver
+          organizationalUnits:
+            - client
+    clientTLS: true # Enable client TLS
+  timeout: 5m
+  apply: IfReady
+```
+
+This example demonstrates how to add TLS to an existing SQL Server cluster using certificates issued by `mssqlserver-ca-issuer`.
+
+Other Supported TLS Actions:
+
+**Remove TLS**: Disable TLS for the SQL Server cluster.
+
+tls Spec:
+```yaml
+tls:
+  remove: true # Remove existing TLS configuration
+```
+
+**Rotate Certificates**: Rotate the TLS certificates used by the SQL Server cluster.
+
+tls Spec:
+```yaml
+tls:
+  rotateCertificates: true # Trigger certificate rotation
+```
+
+**Change the Certificate Issuer**: Update the issuer of the TLS certificate to a new one.
+
+tls Spec:
+```yaml
+tls:
+  issuerRef:
+    name: mssqlserver-new-ca-issuer # New cert-manager issuer name
+    kind: Issuer # Issuer type (Issuer or ClusterIssuer)
+    apiGroup: "cert-manager.io" 
+```
+
+
+### Updates for MSSQLServer CRD
+
+**Removed Fields**:
+
+- `spec.internalAuth`: The `spec.internalAuth` section has been removed. TLS/SSL configurations for internal endpoint authentication of SQL Server Availability Group replicas should now be defined under `spec.tls`.
+
+**Updated Fields**:
+
+- Leader Election Configuration:
+  - The field `spec.leaderElection` is now moved under `spec.topology.availabilityGroup.leaderElection`.
+
+Below is the updated structure reflecting these changes:
+```yaml
+tls:
+  issuerRef:
+    apiGroup: cert-manager.io
+    kind: Issuer
+    name: mssqlserver-ca-issuer
+  certificates:
+    - alias: server
+      emailAddresses:
+        - dev@appscode.com
+      secretName: mssqlserver-server-cert
+      subject:
+        organizationalUnits:
+          - server
+        organizations:
+          - kubedb
+    - alias: client
+      emailAddresses:
+        - abc@appscode.com
+      secretName: mssqlserver-client-cert
+      subject:
+        organizationalUnits:
+          - client
+        organizations:
+          - kubedb
+    - alias: endpoint
+      secretName: mssqlserver-endpoint-cert
+      subject:
+        organizationalUnits:
+          - endpoint
+        organizations:
+          - kubedb
+  clientTLS: true
+topology:
+  availabilityGroup:
+    databases:
+      - agdb1
+      - agdb2
+    leaderElection:
+      electionTick: 10
+      heartbeatTick: 1
+      period: 300ms
+      transferLeadershipInterval: 1s
+      transferLeadershipTimeout: 1m0s
+  mode: AvailabilityGroup
+```
+
+**Additional Notes:**
+- Use `spec.tls` to configure all TLS/SSL-related settings, including endpoint authentication for Availability Group replicas.
+- For leader election, adjust parameters under `spec.topology.availabilityGroup.leaderElection` as needed.
```

Comment:
> Note: Use `spec.tls` to configure all TLS/SSL-related settings, including endpoint authentication for Availability Group replicas.

Remove it:
- For leader election, adjust parameters under `spec.topology.availabilityGroup.leaderElection` as needed.